compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -350,26 +350,18 @@ object Completion {
     *  in the REPL and the IDE.
     */
    private class RenameAwareScope extends Scopes.MutableScope {
-     private[this] val nameToSymbols: mutable.Map[Name, List[Symbol]] = mutable.Map.empty
+     private[this] val nameToSymbols: mutable.Map[SimpleName, List[Symbol]] = mutable.Map.empty
 
      /** Enter the symbol `sym` in this scope, recording a potential renaming. */
      def enter[T <: Symbol](sym: T, name: Name)(implicit ctx: Context): T = {
-       nameToSymbols += name -> (sym :: nameToSymbols.getOrElse(name, Nil))
+       val simpleName = name.stripModuleClassSuffix.toSimpleName
```

Comment:
How come `toSimpleName` is needed ? Aren't we already filtering out all non-simple-names with the completionsFilter ?

Reply:
No, `completionsFilter` filters names based on their kind (`Simple`, `Qualified`, `Numbered` and so on), which is a different distinction from `SimpleName` / `TermName` / `TypeName` as far as I can tell.

We need to do `toSimpleName` here so that `TermName`s and `TypeName`s are grouped.

Reply:
Ah, I see, I would to `toTermName` then since SimpleNames are TermNames anyway, and it means that we don't accidentally mangle encoded names.